ps-modules/MacUtil/MacUtil.psm1

```diff
@@ -0,0 +1,202 @@
+        $dylibName = [System.IO.Path]::GetFileName($libPath)
+
+        Write-Host ($path + "-" + $dylibName)
+
+        if ([System.IO.Path]::GetFileName($path) -eq $dylibName) {
+            Write-Host ">>> Changing ID"
+            $changeCmd = "install_name_tool -id $newPath $libPath"
+        }
+        else {
+            Write-Host ">>> Changing path"
+            $changeCmd = "install_name_tool -change $path $newPath $libPath"
+        }
+
+        Invoke-Expression -Command $changeCmd
+    }
+
+    #Write-Host ">> Final result"
+    #Invoke-Expression -Command $otoolCmd
+}
+
+function ConvertTo-RelativeInstallPaths {
+    param (
+        [string]$directory,
+        [string]$extension
+    )
+
+    $libFiles = Get-ChildItem -Path $directory -Filter "*.$extension"
+    foreach ($libFile in $libFiles) {
+        $filePath = $libFile.FullName
+        $fileName = $libFile.Name
+        Write-Host ">> Processing: $fileName"
+        Set-ItemProperty -Path $filePath -Name IsReadOnly -Value $false
+        Update-LibraryPath -libPath $filePath
+    }
+}
+
+Function Remove-DylibSymlinks {
+    param (
+        [Parameter(Mandatory=$true)][string]$BuildArtifactsPath
+    )
+
+    Push-Location "$BuildArtifactsPath/lib"
+
+    # Enumerate files
+    $main_files = @()
+    $files_with_versions = @()
+    $dylib_files = Get-ChildItem -Path . -Filter "*.dylib"
+    foreach ($dylib_file in $dylib_files) {
+        # Check if the file name contains more than one dot
+        if (-not ($dylib_file.Name -match '\..*\..*')) {
+            $main_files += $dylib_file.Name
+        }
+        else {
+            $files_with_versions += $dylib_file.Name
+        }
+    }
+
+    Write-Host ""
+    Write-Host "Dynamic library dependencies before changes..."
+    foreach($main_file in $main_files) {
+        Write-Host "> $main_file"
+        Invoke-Expression "otool -L '$main_file' | grep '@rpath'"
+    }
+
+    Write-Host ""
+    Write-Host "Updating paths to dynamic dependencies..."
+    foreach ($main_file in $main_files) {
+        # Main file
+        Write-Host ("> $main_file")
+        Invoke-Expression "install_name_tool -id '@rpath/$main_file' '$main_file'"
+
+        # All other files that might point to it
+        foreach ($possible_current_dependency in $files_with_versions) {
+            $base_filename = ($possible_current_dependency -split '[^a-zA-Z0-9]')[0] # Discard anything after the first non-alphanumeric character
+            $new_dependency = "$base_filename.dylib"
+            if ($main_files -contains $new_dependency) {
+                Invoke-Expression "install_name_tool -change '@rpath/$possible_current_dependency' '@rpath/$new_dependency' '$main_file'"
+            } else {
+                Write-Host (">> Matching main file not found for: $possible_current_dependency!!!")
+            }
+        }
+    }
+
+    Write-Host ""
+    Write-Host "Dynamic library dependenies after changes..."
+    foreach($main_file in $main_files) {
```

Comment:
Duplicate code, could extract to a `Write-Host-LIbraryDependencies` function or something.

Reply:
Fixed: e3729cea9d6ca540731b243d618bdacdde5610a1